lib/http/response/body.rb

```diff
@@ -14,20 +14,30 @@ def initialize(client, encoding = Encoding::BINARY)
         @client    = client
         @streaming = nil
         @contents  = nil
-        @encoding  = encoding
+
+        # see issue 312
+        begin
+          @encoding = Encoding.find encoding
+        rescue ArgumentError
+          @encoding = Encoding::BINARY
+        end
       end
 
       # (see HTTP::Client#readpartial)
       def readpartial(*args)
         stream!
-        @client.readpartial(*args)
+        force_encoding @client.readpartial(*args)
```

Comment:
But `#to_s` doesn't have to deal with partial data which ends in the middle of multibyte characters